snapcraft/plugins/catkin.py

```diff
@@ -889,9 +889,11 @@ def find(self, package_name):
 
     def _run(self, arguments):
         with tempfile.NamedTemporaryFile(mode='w+') as f:
-            lines = ['export PYTHONPATH={}'.format(os.path.join(
-                self._catkin_install_path, 'usr', 'lib', 'python2.7',
-                'dist-packages'))]
+            lines = ['export PYTHONPATH={}:{}'.format(
+                os.path.join(self._catkin_install_path, 'usr', 'lib',
+                             'python2.7', 'dist-packages'),
+                os.path.join(self._catkin_install_path, 'opt', 'ros',
+                             'kinetic', 'lib', 'python2.7', 'dist-packages'))]
```

Comment:
I need to do some testing to be sure, but I doesn't sourcing the `setup.sh` as done immediately below this add that to the `PYTHONPATH`? But making this change seems to solve your issue?